chalk-solve/src/clauses/program_clauses.rs

```diff
@@ -120,13 +120,14 @@ impl<I: Interner> ToProgramClauses<I> for AssociatedTyValue<I> {
 }
 
 impl<I: Interner> ToProgramClauses<I> for OpaqueTyDatum<I> {
-    /// Given `opaque type T<..>: A + B = HiddenTy;`, we generate:
+    /// Given `opaque type T<U>: A + B = HiddenTy where U: C;`, we generate:
     ///
     /// ```notrust
-    /// AliasEq(T<..> = HiddenTy) :- Reveal.
-    /// AliasEq(T<..> = !T<..>).
-    /// Implemented(!T<..>: A).
-    /// Implemented(!T<..>: B).
+    /// AliasEq(T<U> = HiddenTy) :- Reveal.
+    /// AliasEq(T<U> = !T<U>).
+    /// WF(T<U>) :- WF(U: C).
+    /// Implemented(!T<U>: A).
+    /// Implemented(!T<U>: B).
```

Comment:
So this seems right, but also sort of...incomplete, in that we never use these where clauses as hypotheses. I guess this is because we would want to do that in the well-formedness checking for the opaque type **definition**, which is not yet implemented?

Reply:
That was my thinking as well. I was thinking those checks would probably belong around [here](https://github.com/rust-lang/chalk/blob/0f43952c643248811a730c2eef63ce2fe60d1a64/chalk-solve/src/wf.rs#L142) based on what we do for ADTs and trait impls. I figure we could save that for a follow-up PR.